src/evo/deterministicmns.h

```diff
@@ -408,14 +412,24 @@ class CDeterministicMNList
 
 private:
     template <typename T>
-    [[nodiscard]] bool AddUniqueProperty(const CDeterministicMN& dmn, const T& v)
+    [[nodiscard]] bool AddUniqueProperty(const CDeterministicMN& dmn, const T& v, std::optional<bool> specific_legacy_bls_scheme = std::nullopt)
     {
         static const T nullValue;
         if (v == nullValue) {
             return false;
         }
 
-        auto hash = ::SerializeHash(v);
+        uint256 hash;
+        if constexpr (std::is_same<T, CBLSLazyPublicKey>()) {
+            assert(specific_legacy_bls_scheme.has_value());
+            CBLSPublicKey local_pub_key = v.Get();
+            CBLSPublicKeyVersionWrapper wrapped_pub_key(local_pub_key, specific_legacy_bls_scheme.value());
+            hash = ::SerializeHash(wrapped_pub_key);
+        } else {
+            assert(!specific_legacy_bls_scheme.has_value());
```

Comment:
This must be removed or changed to:
```suggestion
if (specific_legacy_bls_scheme.has_value()) assert(!specific_legacy_bls_scheme.value();
```